pkg/image/fetcher.go

```diff
@@ -120,7 +137,17 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 		return nil, err
 	}
 
-	return f.fetchDaemonImage(name)
+	image, err := f.fetchDaemonImage(name)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update image pull record in the JSON file
+	if err := f.updateImagePullRecord(name, time.Now().Format(time.RFC3339)); err != nil {
```

Comment:
Do we want to keep track of every pulling even if the **interval policy** is not enable?

Reply:
I also thought about this, and I think keeping the pulling track even when interval policy is not enable will be good. cause let's suppose user enable it with some x days, and user might thought as the image pull recently(when interval policy was not enabled) it will not pull again, but if we don't track pulling then it will pull again. I mean the effect of policy will start after one pull of each image.

Reply:
My only concern is to break some current behaivor for end-users and they don't know why, let's suppose for some reason we introduced a bug with the interval policy code, maybe with the hard drive space on environments that is limited. 
It could affect the users and they don't even using the feature, that's why I like to keep things as they are and only if they enable the feature we start keeping track of the data.

Reply:
Hmm, that's a fair point, thank you for addressing it.

Reply:
@jjbustamante  should I go for the test cases now? or wait some more time to get reviews from you or other maintainers?

Reply:
I think you can start working in your test coverage yes.